lib/services/ngsi/entities-NGSI-v2.js

```diff
@@ -552,6 +505,31 @@ function sendUpdateValueNgsi2(entityName, measures, typeInformation, token, call
             let e = {};
             e.id = String(ename);
             e.type = String(etype);
+            let timestamp = { type: constants.TIMESTAMP_TYPE_NGSI2 }; //timestamp scafold-attr for insertions.
+            let timestampAttrs = null;
+            if (mustInsertTimeInstant) {
+                // get timestamp for current entity
+
+                timestampAttrs = entities[ename][etype].filter((item) => item.name === constants.TIMESTAMP_ATTRIBUTE);
+                if (timestampAttrs && timestampAttrs.length > 0) {
+                    timestamp.value = timestampAttrs[0]['value'];
+                }
+
+                if (timestamp.value) {
+                    if (!moment(timestamp.value, moment.ISO_8601, true).isValid()) {
+                        callback(new errors.BadTimestamp(timestamp.value, entityName, typeInformation));
+                        return;
+                    }
+                } else {
+                    if (!typeInformation.timezone) {
```

Comment:
Maybe this could be precalculated to simplify logic and optimize. Minor.

Reply:
Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1593/commits/4c8a62478a645ff66218e73012b3b52c0c9da833